str/qc/qc_annotator.py

```diff
@@ -0,0 +1,223 @@
+    )
+    mt = mt.annotate_entries(
+        allele_2_bp_length=mt.allele_2_rep_length * mt.motif_length
+    )
+
+    # annotate rows with counts of alleles that appear in each VARID
+    ht = mt.select_rows(
+        alleles_rep_lengths=hl.agg.collect(mt.allele_1_rep_length).extend(
+            hl.agg.collect(mt.allele_2_rep_length)
+        )
+    ).rows()
+
+    # explode the allele_array to create one row for each element in the array
+    exploded_table = ht.explode(ht.alleles_rep_lengths)
+
+    # aggregate the counts for each distinct value in the exploded allele_array
+    aggregated_table = exploded_table.group_by(exploded_table.REPID).aggregate(
+        allele_array_counts=hl.agg.counter(exploded_table.alleles_rep_lengths)
+    )
+
+    # finds the mode allele at each locus
+    max_key_expr = hl.bind(
+        lambda d: hl.bind(
+            lambda kv: kv[0], hl.sorted(d.items(), key=lambda kv: -kv[1])[0]
+        ),
+        aggregated_table.allele_array_counts,
+    )
+    aggregated_table = aggregated_table.annotate(mode_allele=max_key_expr)
+
+    # annotate the mode allele and dictionary of frequency of each allele into mt
+    mt = mt.annotate_rows(aggregated_info=aggregated_table[mt.REPID])
+    # number of distinct alleles identified at each locus
+    mt = mt.annotate_rows(
+        num_alleles=hl.len(mt.aggregated_info.allele_array_counts.values())
+    )
+
+    # the difference between each allele and the mode allele
+    mt = mt.annotate_entries(
+        allele_1_minus_mode=mt.allele_1_rep_length - mt.aggregated_info.mode_allele
+    )
+    mt = mt.annotate_entries(
+        allele_2_minus_mode=mt.allele_2_rep_length - mt.aggregated_info.mode_allele
+    )
+
+    # annotate the sum of alleles that are not the mode allele
+    mt = mt.annotate_entries(
+        allele_1_is_non_mode=hl.if_else(mt.allele_1_minus_mode != 0, True, False)
+    )
+    mt = mt.annotate_entries(
+        allele_2_is_non_mode=hl.if_else(mt.allele_2_minus_mode != 0, True, False)
+    )
+    mt = mt.annotate_rows(
+        sum_allele_1_is_not_mode=hl.agg.sum(hl.cond(mt.allele_1_is_non_mode, 1, 0)),
+        sum_allele_2_is_not_mode=hl.agg.sum(hl.cond(mt.allele_2_is_non_mode, 1, 0)),
+    )
+    mt = mt.annotate_rows(
+        sum_alleles_is_not_mode=mt.sum_allele_1_is_not_mode
+        + mt.sum_allele_2_is_not_mode
+    )
+    mt = mt.drop('sum_allele_1_is_not_mode', 'sum_allele_2_is_not_mode')
+
+    # proportion of alleles that are not the mode allele
+    mt = mt.annotate_rows(
+        prop_alleles_is_not_mode=mt.sum_alleles_is_not_mode
+        / hl.sum(mt.aggregated_info.allele_array_counts.values())
+    )
+
+    # add in Gymrek binomial HWEP implementation (Hail Query's HWEP implementation only works for biallelic loci)
+    # https://github.com/gymrek-lab/TRTools/blob/master/trtools/utils/utils.py#L325
+    mt = mt.annotate_rows(n_hom=mt.variant_qc.n_called - mt.variant_qc.n_het)
```

Comment:
Hom counts are in the variant_qc struct as `mt.variant_qc.homozygote_count`, an array of ints for `[ref, alt1, [alt2...]]`

Reply:
if you want the pure hom counts you can probably do a hl.sum or hl.agg.sum on the array